pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -577,8 +578,7 @@ public PinotResourceManagerResponse rebuildBrokerResourceFromHelixTags(@Nonnull
     Set<String> idealStateBrokerInstances = brokerIdealState.getInstanceSet(tableNameWithType);
 
     if (idealStateBrokerInstances.equals(brokerInstances)) {
-      return new PinotResourceManagerResponse(
-          "Broker resource is not rebuilt because ideal state is the same for table {} " + tableNameWithType, false);
+      throw new BadRequestException("Broker resource is not rebuilt because ideal state is the same for table: " + tableNameWithType);
```

Comment:
this is not a bad request. It is an idempotent operation, so why throw exception ? why not return success?